pkg/providers/aws/cluster_vpc.go

```diff
@@ -457,37 +457,53 @@ func getSubnets(ec2Svc ec2iface.EC2API) ([]*ec2.Subnet, error) {
 	return subs, nil
 }
 
+func getVPCIDByClusterSubnets(ec2Svc ec2iface.EC2API, clusterID string) (*string, error) {
+	var subs []*ec2.Subnet
+	listOutput, err := ec2Svc.DescribeSubnets(&ec2.DescribeSubnetsInput{})
+	if err != nil {
+		return aws.String(""), err
+	}
+	subs = listOutput.Subnets
+	for _, sub := range subs {
+		for _, tag := range sub.Tags {
+			if (*tag.Value == "owned" || *tag.Value == "shared") &&
+				*tag.Key == fmt.Sprintf("%s%s", clusterOwnedTagKeyPrefix, clusterID) {
+				return sub.VpcId, nil
+			}
+		}
+	}
+	return aws.String(""), errorUtil.New(fmt.Sprintf("failed to get cluster vpc, no vpc found: error getting vpc id from associated subnets: could not find cluster associated subnets with clusterID %s", clusterID))
+}
+
 // function to get vpc of a cluster
 func getClusterVpc(ctx context.Context, c client.Client, ec2Svc ec2iface.EC2API, logger *logrus.Entry) (*ec2.Vpc, error) {
 	// first call to aws api from the network provider is to get cluster vpc
 	// polling to allow credential minter time to reconcile credentials
-	vpcs, err := ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{})
-	if err != nil {
-		return nil, errorUtil.Wrap(err, "error getting vpcs")
-	}
 
 	// get cluster id
 	clusterID, err := resources.GetClusterID(ctx, c)
 	if err != nil {
 		return nil, errorUtil.Wrap(err, "error getting clusterID")
 	}
 
-	// find associated vpc to cluster
-	var foundVPC *ec2.Vpc
-	logger.Infof("searching for cluster %s vpc", clusterID)
-	for _, vpc := range vpcs.Vpcs {
-		for _, tag := range vpc.Tags {
-			if *tag.Value == fmt.Sprintf("%s-vpc", clusterID) {
-				foundVPC = vpc
-			}
-		}
+	vpcId, err := getVPCIDByClusterSubnets(ec2Svc, clusterID)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error getting vpc id from associated subnets")
 	}
 
-	if foundVPC == nil {
+	vpcs, err := ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{VpcIds: []*string{vpcId}})
```

Comment:
I wonder could this be changed to something like `ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{VpcIds: []aws.string{vpcId}})`

Reply:
hey, Ciran, Is tthis what you meant? `vpcs, err := ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{VpcIds: []*string{aws.String(*vpcId)}})
`
It feels odd :thinking: